### DISCUSSION TITLE
Introducing Schemar - a GitHub Action for validating structured data

### DISCUSSION THREAD

**Discussion by johnnyreilly** [schemaorg/schemaorg#3432] [Show and tell]
Hey everyone!

I'm a longtime user of the marvellous https://validator.schema.org/ 

Over Christmas I got to thinking that I'd like the manual structured data checks I've been doing on my sites to be automated.

I've built [Schemar](https://github.com/marketplace/actions/schemar-ci-action) to that end; it makes use of the [Schema Markup Validator](https://validator.schema.org/) and surfaces the results in GitHub Action.  It fails if structured data is missing or invalid.

What do you think?  My hope is that this can help the web build websites that surface valid structured data. I've written a blog post with more details here: https://johnnyreilly.com/schemar-github-action-to-validate-structured-data

**Comment by KalleOlaviNiemitalo** [comment]
<https://github.com/schemaorg/schemaorg/issues/3013#issuecomment-1063725012> notes that validator.schema.org started blocking requests.  Is it likely to block all requests from GitHub-hosted runners if the GitHub action is run too frequently?

I don't see any restrictions against automated use in <https://webschemas.org/docs/validator.html> or <https://webschemas.org/docs/terms.html>, though.

**Reply by johnnyreilly** [reply]
That's a great question! I used it a bunch in testing and never had any issues. That's one of the reasons for mentioning it here - essentially I want to build something that helps the community; it would be unfortunate if it were blocked. Does anyone know who looks after the underlying system?
